pandas/core/series.py

```diff
@@ -5579,6 +5579,15 @@ def _align_for_op(self, right, align_asobject: bool = False):
             # avoid repeated alignment
             if not left.index.equals(right.index):
                 if align_asobject:
+                    if left.dtype != np.bool_ and left.dtype != np.bool_:
+                        warnings.warn(
+                            "Operation between non boolean Series with different "
+                            "indexes will no longer return a boolean result in "
+                            "a future version. Cast both Series to object type "
+                            "to maintain the prior behavior.",
+                            FutureWarning,
+                            stacklevel=find_stack_level(),
+                        )
```

Comment:
FWIW i think we should deprecate this casting in all cases

Reply:
Agreed, I just wanted to issue to warning though were the result type would change i.e. is a user is doing logical ops on booleans they should already get a boolean result hence no need to warn

Reply:
so do we need to do a second deprecation to handle more cases or when this deprecation is enforced will we be able to remove the entire align_asobject block?